### PR TITLE
Several fixes to pkg-config dependencies and the pkg-config module

### PR DESCRIPTION
**Pull request by nirbheek**
`dependency(..., static: true)` had little effect when using pkg-config dependencies because we still passed `-lfoo -lbaz` to the compiler which with gcc meant that you would use shared libraries, and with MSVC meant that we would convert that to `foo.lib baz.lib` which meant that we would use import libraries (when linking to projects built with Meson) or possibly static libraries (those that were not built with Meson, because we name them `libfoo.a`).

With these changes, when `static: true` is set, we parse the pkg-config output, and if it contains `-L` fields, we use `cc.find_library`'s manual library searching feature to find the actual location of the library and use it directly.

These changes also improve `cc.find_library`'s manual searching to only search for platform-specific libraries. Earlier, it would find `libfoo.dylib` as a valid file on non-macOS platforms.

You can now also specify what type of library you want to find. For instance, you can specify that you only want static libraries or only shared libraries or prefer static over shared, etc. This functionality is currently not exposed to build files because it only works when `extra_dirs` is specified. However, I can see it being useful in the future because this is an oft-asked-for feature among autotools and libtool users.

We now also use `shlex.split` to parse `Libs:` and `Cflags:` in pkg-config files so that we properly parse them when `libdir`, `prefix`, etc contain spaces. I also found that our pkg-config generator does not properly escape paths with spaces (`prefix`, `libdir`, etc), which is [required by all implementations](https://github.com/pkgconf/pkgconf/issues/153), so I also fixed that.

Overall, pkg-config usage on Windows should be much better with these commits.

### CODE REVIEW

File: run_unittests.py [hunk 1]
```diff
@@ -1492,6 +1564,8 @@ def assertMesonOutputs(self, contents, match, extra_args=None, langs=None):
     def test_dependency(self):
         if not shutil.which('pkg-config'):
             raise unittest.SkipTest('pkg-config not found')
+        if subprocess.call(['pkg-config', '--exists', 'zlib']) != 0
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__

SyntaxError: invalid syntax

File: run_unittests.py [hunk 2]
```diff
@@ -1492,6 +1564,8 @@ def assertMesonOutputs(self, contents, match, extra_args=None, langs=None):
...
+            raise unittest.SkipTest('zlib not found with pkg-config')
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__

unexpected indentation

**Comment by nirbheek** [comment]
This is strange. Neither appveyor nor Travis are running on this PR now.

**Comment by jpakkane** [comment]
Maybe if this is closed and then a new MR is created it will work?

**Comment by nirbheek** [comment]
Let's try that.

**Comment by nirbheek** [comment]
Finally worked: https://github.com/mesonbuild/meson/pull/2611

**Pull request closed without merging**